File: internal/ui/styles.go
package ui

import (
	"os"

	"github.com/charmbracelet/lipgloss"
	"github.com/muesli/termenv"
)

// Semantic styles used across the CLI. These mirror the previous fatih/color
// palette but are Lip Gloss styles rendered via .Render().
var (
	Success = lipgloss.NewStyle().Foreground(lipgloss.Color("2"))
	Error   = lipgloss.NewStyle().Foreground(lipgloss.Color("1"))
	Warning = lipgloss.NewStyle().Foreground(lipgloss.Color("3"))
	Info    = lipgloss.NewStyle().Foreground(lipgloss.Color("6"))

	Bold      = lipgloss.NewStyle().Bold(true)
	Header    = lipgloss.NewStyle().Foreground(lipgloss.Color("6")).Bold(true)
	Hostname  = lipgloss.NewStyle().Foreground(lipgloss.Color("3")).Bold(true)
	Secondary = lipgloss.NewStyle().Foreground(lipgloss.Color("8"))
)

func init() {
	// Disable color when explicitly requested (NO_COLOR is the standard,
	// HLADMIN_NO_COLOR is a tool-specific override). Lip Gloss already downgrades
	// to no color when stdout is not a TTY.
	if os.Getenv("NO_COLOR") != "" || os.Getenv("HLADMIN_NO_COLOR") != "" {
		lipgloss.SetColorProfile(termenv.Ascii)
	}
}

// IsTerminal reports whether stdout is connected to a terminal.
func IsTerminal() bool {
	fileInfo, err := os.Stdout.Stat()
	if err != nil {
		return false
	}
	return (fileInfo.Mode() & os.ModeCharDevice) != 0
}

File: src/ui/styles.rs
use console::Style;
use std::io::IsTerminal;

/// Semantic styles used across the CLI (ANSI 16-color palette).
pub fn success() -> Style {
    Style::new().green()
}
pub fn error() -> Style {
    Style::new().red()
}
pub fn warning() -> Style {
    Style::new().yellow()
}
pub fn info() -> Style {
    Style::new().cyan()
}
pub fn bold() -> Style {
    Style::new().bold()
}
pub fn header() -> Style {
    Style::new().cyan().bold()
}
pub fn hostname() -> Style {
    Style::new().yellow().bold()
}
pub fn secondary() -> Style {
    Style::new().black().bright()
}

/// Decides color output once at startup: colors are emitted only when stdout is
/// a TTY and neither NO_COLOR (the standard) nor HLADMIN_NO_COLOR (tool-specific
/// override) is set to a non-empty value.
pub fn init() {
    let disabled_by_env = |name: &str| std::env::var_os(name).is_some_and(|v| !v.is_empty());
    let enabled = std::io::stdout().is_terminal()
        && !disabled_by_env("NO_COLOR")
        && !disabled_by_env("HLADMIN_NO_COLOR");
    console::set_colors_enabled(enabled);
}

/// Reports whether stdout is connected to a terminal.
pub fn is_terminal() -> bool {
    std::io::stdout().is_terminal()
}
